src/App.tsx

```diff
@@ -1,10 +1,61 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-import { USER_ID } from './api/todos';
+import { addTodos, getTodos, USER_ID } from './api/todos';
+import { TodoList } from './components/TodoList';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
+import { ErrorMessage } from './components/ErrorMessage';
+import { TodoForm } from './components/TodoForm';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [renderCount, setRenderCount] = useState(0);
+
+  const increaseCounter = () => {
+    setRenderCount(prevCount => prevCount + 1);
+  };
+
+  const addTodo = async (todo: Todo): Promise<void> => {
+    try {
+      const newTodo = await addTodos(todo);
+
+      setTodos(prevTodos => [...prevTodos, newTodo]);
+    } catch (error) {
+      setErrorMessage('Unable to add a todo');
+      throw error;
+    }
+  };
+
+  useEffect(() => {
+    let filteredTodos: Todo[];
+
+    getTodos()
+      .then(todosFromServer => {
+        switch (filter) {
+          case 'active':
+            filteredTodos = todosFromServer.filter(
+              todo => todo.completed === false,
+            );
+            break;
+          case 'completed':
+            filteredTodos = todosFromServer.filter(
+              todo => todo.completed === true,
+            );
+            break;
+          default:
+            filteredTodos = todosFromServer;
+            break;
+        }
+
```

Comment:
you don't need to filter the array here, just save the todosFromServer in todos state and create some var in component
```
const filteredTodos = getFilteredTodos()
```
move switch case into the getFilteredTodos function